### PR TITLE
Initialization check for methods with non-hot parameters

### PR DESCRIPTION
**Pull request by natsukagami**
## Description

- Add a `@nonHotParameters` annotation
- Change thisV mostly to Value
- Handle @nonHotParameter calls for Ref case

### CODE REVIEW

File: tests/init/pos/case-class-apply.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+case class Case(b: Base)
```
Comment (natsukagami) [inline tests/init/pos/case-class-apply.scala:1]: This tests the `Hot` case in the `call` method but not `Ref`. How do I add such a case?

**Comment by natsukagami** [comment]
`sbt scala3-compiler-bootstrapped/compile` output with `-Ysafe-init` with this PR on: [output.log](https://github.com/lampepfl/dotty/files/7634621/output.log)

**Comment by natsukagami** [comment]
With previous commit applied: [output.log](https://github.com/lampepfl/dotty/files/7635014/output.log)
Number of warnings went down to 46! (from 57 without the PR: [output-2.log](https://github.com/lampepfl/dotty/files/7635021/output-2.log))

**Comment by liufengyun** [comment]
This is now superseded by https://github.com/lampepfl/dotty/pull/14283.

**Pull request closed without merging**